src/gei/Commands/GenerateArchiveCommand.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class GenerateArchiveCommand : Command
+    {
+        private readonly OctoLogger _log;
+        private readonly ISourceGithubApiFactory _sourceGithubApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+
+        private const int _timeoutInHours = 10;
+        private const int _delayInMS = 10000; // 10 seconds
+
+        public GenerateArchiveCommand(OctoLogger log, ISourceGithubApiFactory sourceGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider) : base("generate-archive")
+        {
+            _log = log;
+            _sourceGithubApiFactory = sourceGithubApiFactory;
+            _environmentVariableProvider = environmentVariableProvider;
+
+            Description = "Invokes the GitHub Migration API's to generate a migration archive";
+            Description += Environment.NewLine;
+            Description += "Note: Expects GH_PAT and GH_SOURCE_PAT env variables to be set. GH_SOURCE_PAT is optional, if not set GH_PAT will be used instead. This authenticates to the source GHES API.";
```

Comment:
This should include that it expects `--ghes-url` as well

Reply:
--ghes-url is treated the same as any other arg, it will show up in the help in the list of args. We call out the PAT's separately there because they aren't args, but env vars.